include/zephyr/rtio/rtio.h

```diff
@@ -423,6 +447,14 @@ static inline void rtio_sqe_prep_read(struct rtio_sqe *sqe,
 	sqe->userdata = userdata;
 }
 
+/**
+ * @brief Prepare a read op submission with context's mempool
+ *
+ * @see rtio_sqe_prep_read()
+ */
+#define rtio_sqe_read_with_pool(sqe, iodev, prio, userdata)                                        \
```

Comment:
any reason to not use static inline here?

It seems reasonable to also set the SQE_MEMPOOL_FLAG here